src/routes/recovery/guards/enable-recovery-alerts.guard.ts

```diff
@@ -67,5 +69,17 @@ export class EnableRecoveryAlertsGuard implements CanActivate {
       this.loggingService.debug(e);
       return false;
     }
+
+    try {
+      const { safes } = await this.safeRepository.getSafesByModule({
+        chainId,
+        moduleAddress,
+      });
+
+      return safes.some((safe) => getAddress(safe) === getAddress(safeAddress));
+    } catch (e) {
+      this.loggingService.debug(e);
+      return false;
```

Comment:
Same here 🙏 

https://github.com/safe-global/safe-client-gateway/pull/1267#discussion_r1519432903

Reply:
Merged in db117f82d687f40dcc2a002ba2e3b4c8c7faf449.